File: server/src/collectors/jcabNotams.js
/**
 * JCAB NOTAMs — Japan Civil Aviation Bureau notices to airmen.
 * https://aim-naviserv.mlit.go.jp/aimjp/web/notam.html
 *
 * NOTAMs surface VIP-state-visit TFRs, naval-exercise restricted zones,
 * disaster-response no-fly zones — often *before* news. The portal is an
 * HTML form that requires a POST to retrieve the actual NOTAM listing, so
 * we record portal status + one rolling intel item; the body can be
 * fleshed out with a Playwright pass later.
 */

import { intelEnvelope, intelUid } from '../utils/intelHelpers.js';
import { fetchHead } from './_liveHelpers.js';

const SOURCE_ID = 'jcab-notams';
const PROBE_URL = 'https://aim-naviserv.mlit.go.jp/aimjp/web/notam.html';

export default async function collectJcabNotams() {
  const live = await fetchHead(PROBE_URL);
  const items = [{
    uid: intelUid(SOURCE_ID, 'notam-portal'),
    title: 'JCAB NOTAMs (notices to airmen)',
    summary: 'Japan Civil Aviation Bureau NOTAM portal — temporary flight restrictions, exercise areas, VIP transit corridors',
    link: PROBE_URL,
    language: 'ja',
    published_at: new Date().toISOString(),
    tags: ['aviation', 'notam', 'tfr', 'mlit', live ? 'reachable' : 'unreachable'],
    properties: { operator: 'MLIT JCAB AIM Naviservice', reachable: live },
  }];

  return intelEnvelope({
    sourceId: SOURCE_ID,
    items,
    live,
    description: 'JCAB NOTAMs — notices to airmen (TFRs, exercise zones, VIP transit)',
  });
}

File: native/collectors/sources/jcab_notams.c
/* collectors/transport/sources/jcab_notams.c
 * Port of server/src/collectors/jcabNotams.js — fetchHead reachability
 * probe → ONE intel item (uid jcab-notams|notam-portal). OTHER
 * portal-status family (lib/probe.h). */
#include "../../source.h"
#include "../../lib/probe.h"
#include "../../third_party/cJSON.h"
#include <stdio.h>
#include <stdlib.h>

#define PROBE_URL "https://aim-naviserv.mlit.go.jp/aimjp/web/notam.html"

static int run(const source_ctx *ctx, intel_sink *sink) {
  int live = probe_head(ctx->http, PROBE_URL);
  char now[32]; probe_iso_now(now, sizeof now);

  /* tags: ['aviation','notam','tfr','mlit', live?'reachable':'unreachable'] */
  cJSON *tags = cJSON_CreateArray();
  cJSON_AddItemToArray(tags, cJSON_CreateString("aviation"));
  cJSON_AddItemToArray(tags, cJSON_CreateString("notam"));
  cJSON_AddItemToArray(tags, cJSON_CreateString("tfr"));
  cJSON_AddItemToArray(tags, cJSON_CreateString("mlit"));
  cJSON_AddItemToArray(tags, cJSON_CreateString(live ? "reachable" : "unreachable"));
  char *tj = cJSON_PrintUnformatted(tags);

  /* properties: { operator, reachable } — EXACT JS key order */
  cJSON *p = cJSON_CreateObject();
  cJSON_AddStringToObject(p, "operator", "MLIT JCAB AIM Naviservice");
  cJSON_AddBoolToObject(p, "reachable", live);
  char *pj = cJSON_PrintUnformatted(p);

  intel_item it = {0};
  it.remote_key   = "notam-portal";           /* → uid jcab-notams|notam-portal */
  it.title        = "JCAB NOTAMs (notices to airmen)";
  it.summary      = "Japan Civil Aviation Bureau NOTAM portal — temporary flight restrictions, exercise areas, VIP transit corridors";
  it.link         = PROBE_URL;
  it.lang         = "ja";
  it.published_at = now;
  it.tags_json    = tj;
  it.properties_json = pj;
  int rc = sink->emit(sink, &it);

  free(tj); free(pj);
  cJSON_Delete(tags); cJSON_Delete(p);
  fprintf(stderr, "[jcab-notams] probe reachable=%d\n", live);
  return rc >= 0 ? 0 : -1;
}

static const source_def jcab_notams_def = {
  .id = "jcab-notams", .collector = "transport",
  .name = "JCAB NOTAMs", .name_ja = "JCAB NOTAM",
   .update_interval_sec = 1800, .run = run };
REGISTER_SOURCE(jcab_notams_def)
